File: Ruby/ruby/RPG/samurai.rb
require_relative "human"

class Samurai < Human 
    @@how_many = 0
    def initialize
        @health = 200
        @@how_many += 1
    end 

    def death_blow(target)
        target.health = 0
        self 
    end 
    def meditate 
        @health = 200
        self 
    end 

    def self.count
        @@how_many
    end 
end 

s1 = Samurai.new 
s2 = Samurai.new 
puts Samurai.count 

File: Java/Spring/ObjectMaster/src/com/objectmaster/models/Samurai.java
package com.objectmaster.models;

public class Samurai extends Human {
	public static int count = 0;
	
	public Samurai() {
		count++;
		this.health = 200;
	}
	
	public void deathBlow(Human target) {
		target.health = 0;
		this.health /= 2;
	}
	
	public void meditate() {
		this.health += (this.health/2);
	}
	
	public static void howMany() {
		System.out.println(count); 
	}
}
